File: src/main/java/com/mini/credit/service/OperationEpargneService.java
package com.mini.credit.service;

import com.mini.credit.dto.epargne.OperationEpargneRequest;
import com.mini.credit.dto.epargne.OperationEpargneResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface OperationEpargneService {
    OperationEpargneResponse enregistrer(OperationEpargneRequest request);
    List<OperationEpargneResponse> getByCompte(Long compteId);
    List<OperationEpargneResponse> getByMembre(Long membreId);
    List<OperationEpargneResponse> getAll();  // Keep for backward compatibility
    Page<OperationEpargneResponse> getAll(Pageable pageable);  // PHASE 3B: New paginated version
    boolean isCurrentUserAccount(Long compteId);
    boolean isCurrentUserMembre(Long membreId);
}

File: frontend/src/app/features/epargne/services/operation-epargne.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../../../core/services/api.config';
import { OperationEpargneRequest } from '../models/operation-epargne-request';
import { OperationEpargneResponse } from '../models/operation-epargne-response';
import { Page } from '../../../shared/models/page.model';


@Injectable({
  providedIn: 'root'
})
export class OperationEpargneService {
  private http = inject(HttpClient);
  private readonly apiUrl = `${API_BASE_URL}/operations-epargne`;

  enregistrer(request: OperationEpargneRequest): Observable<OperationEpargneResponse> {
    return this.http.post<OperationEpargneResponse>(this.apiUrl, request);
  }

  // PHASE 3B: Paginated version
  getAllPaginated(page: number = 0, size: number = 10): Observable<Page<OperationEpargneResponse>> {
    return this.http.get<Page<OperationEpargneResponse>>(`${this.apiUrl}?page=${page}&size=${size}`);
  }

  // Backward compatibility: keep original method
  getAll(): Observable<OperationEpargneResponse[]> {
    return this.http.get<OperationEpargneResponse[]>(this.apiUrl);
  }

  getByCompte(compteId: number): Observable<OperationEpargneResponse[]> {
    return this.http.get<OperationEpargneResponse[]>(`${this.apiUrl}/compte/${compteId}`);
  }

  getByMembre(membreId: number): Observable<OperationEpargneResponse[]> {
    return this.http.get<OperationEpargneResponse[]>(`${this.apiUrl}/membre/${membreId}`);
  }
}